File: dp_labs_and_solutions/dp_labs_and_solutions/dp_java_labs_and_solutions/data_structures/dp_container.java
// dp_container.java

public abstract class dp_container<T>
{
	public abstract dp_iterator<T> make_iterator();

	public int get_size()
	{
		int result = 0;
		dp_iterator<T> a_iterator = make_iterator();
		while (!a_iterator.is_done())
		{
			result++;
			a_iterator.advance();
		}
		return result;
	}
}

File: dp_labs_and_solutions/original/dp_labs_and_solutions/dp_matlab_labs_and_solutions/visitor/solution/DpContainer.m
% DpContainer.m

classdef (Abstract) DpContainer < handle
	methods (Abstract)
		result = makeIterator(theDpContainer)
	end

	methods
		function result = getSize(theDpContainer)
			aIterator = theDpContainer.makeIterator;
			result = 0;
			while ~aIterator.isDone
				result = result + 1;
				aIterator.advance;
			end
		end
	end
end
